conditions/operator_conditions_test.go

```diff
@@ -0,0 +1,338 @@
+			conditionStatusFor = conditionFoo
+			operatorCondition.Status.Conditions[0].Status = metav1.ConditionUnknown
+
+			// IsConditionStatusTrue should return false
+			val, err := IsConditionStatusTrue(operatorCondition, conditionStatusFor)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(val).To(BeFalse())
+
+			// IsConditionStatusFalse should return false
+			val, err = IsConditionStatusFalse(operatorCondition, conditionStatusFor)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(val).To(BeFalse())
+
+			// IsConditionStatusUnknown should return true
+			val, err = IsConditionStatusUnknown(operatorCondition, conditionStatusFor)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(val).To(BeTrue())
+
+		})
+		It("should error when condition cannot be found", func() {
+			conditionStatusFor = conditionBar
+
+			// IsConditionStatusTrue should return error
+			val, err := IsConditionStatusTrue(operatorCondition, conditionStatusFor)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("%s not found", conditionStatusFor)))
+			Expect(val).To(BeFalse())
+
+			// IsConditionStatusFalse should return error
+			val, err = IsConditionStatusFalse(operatorCondition, conditionStatusFor)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("%s not found", conditionStatusFor)))
+			Expect(val).To(BeFalse())
+
+			// IsConditionStatusUnknown should return error
+			val, err = IsConditionStatusUnknown(operatorCondition, conditionStatusFor)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("%s not found", conditionStatusFor)))
+			Expect(val).To(BeFalse())
+		})
+	})
+
+	Describe("IsConditionStatusPresentAndEqual", func() {
+		var (
+			conditionType   string
+			conditionStatus metav1.ConditionStatus
+		)
+
+		It("should return true when condition is in the specified status", func() {
+			conditionType = conditionFoo
+			conditionStatus = metav1.ConditionTrue
+
+			val, err := IsConditionStatusPresentAndEqual(operatorCondition, conditionType, conditionStatus)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(val).To(BeTrue())
+		})
+
+		It("should return false when condition is not present in the specified status", func() {
+			conditionType = conditionFoo
+			conditionStatus = metav1.ConditionUnknown
+
+			val, err := IsConditionStatusPresentAndEqual(operatorCondition, conditionType, conditionStatus)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(val).To(BeFalse())
+		})
+
+		It("should return error when condition is not present", func() {
+			conditionType = conditionBar
+			conditionStatus = metav1.ConditionTrue
+
+			val, err := IsConditionStatusPresentAndEqual(operatorCondition, conditionType, conditionStatus)
+			Expect(err).To(HaveOccurred())
+			Expect(val).To(BeFalse())
+		})
+
+	})
+
+	Describe("GetNamespacedName", func() {
+		It("should error when name of the operator condition cannot be found", func() {
```

Comment:
The only nit I have about the test is the order doesn't match the order of the methods in the operator_conditions.go file, so it was hard to review since I had to search for the tests for each method. I won't hold this PR up for this but just in general it makes it easier to review.